packages/fx-core/src/plugins/solution/fx-solution/v2/deploy.ts

```diff
@@ -37,17 +45,25 @@ export async function deploy(
   envInfo: v2.DeepReadonly<v2.EnvInfoV2>,
   tokenProvider: TokenProvider
 ): Promise<Result<Void, FxError>> {
+  ctx.telemetryReporter?.sendTelemetryEvent(SolutionTelemetryEvent.DeployStart, {
+    [SolutionTelemetryProperty.Component]: SolutionTelemetryComponentName,
+    [SolutionTelemetryProperty.SkipAadDeploy]: inputs.skipAadDeploy ?? "yes",
+  });
   const provisionOutputs: Json = envInfo.state;
   const inAzureProject = isAzureProject(getAzureSolutionSettings(ctx));
   const provisioned = provisionOutputs[GLOBAL_CONFIG][SOLUTION_PROVISION_SUCCEEDED] as boolean;
 
   if (inAzureProject && !provisioned) {
     return err(
-      new UserError(
-        SolutionSource,
-        SolutionError.CannotDeployBeforeProvision,
-        getLocalizedString("core.NotProvisionedNotice", ctx.projectSetting.appName),
-        getLocalizedString("core.NotProvisionedNotice", ctx.projectSetting.appName)
+      sendErrorTelemetryThenReturnError(
+        SolutionTelemetryEvent.Deploy,
+        new UserError(
+          SolutionSource,
+          SolutionError.CannotDeployBeforeProvision,
+          getLocalizedString("core.NotProvisionedNotice", ctx.projectSetting.appName),
```

Comment:
Use localized string for displayMessage and default string for message(so we can analyze telemetry with English version string)?